src/main/java/com/gs/crdtools/SourceGenFromSpec.java

```diff
@@ -71,38 +71,41 @@ static void toZip(Map<Path, String> content, Path output) throws IOException {
      * @throws IOException If any error occurs while loading the given paths.
      */
-    public static Map<Path, String> generateSourceCodeFromSpecs(String specs) throws IOException {
+    public static Map<Path, String> generateSourceCodeFromSpecs(List<Spec> specs) throws IOException {
         // setting this system property has the interesting effect of preventing the
         // generation of a whole set of unrelated files that we don't care about.
         System.setProperty("generateModels", "true");
 
+        HashMap <Path, String> result = HashMap.empty();
         var tmpOutputDir = Files.createTempDirectory("openAPIGen");
 
-        var cc = new CodegenConfigurator()
-                .setInputSpec(specs)
-                .setLang(CrdtoolsCodegen.class.getCanonicalName())
-                .setOutputDir(tmpOutputDir.toAbsolutePath().toString())
-                .setModelPackage("com.gs.crdtools")
-                // CodegenConfigurator modifies its Map arguments, so we need to wrap it in something mutable
-                .setAdditionalProperties(
-                    HashMap.of(
-                        "java8", (Object)true,
-                        "hideGenerationTimestamp", true,
-                        "notNullJacksonAnnotation", true
-                    ).toJavaMap()
-                )
-                .setTypeMappings(
-                    HashMap.of(
-                        V1ObjectMeta.class.getSimpleName(),
-                        V1ObjectMeta.class.getCanonicalName()
-                    ).toJavaMap()
-                );
-        try {
+        for (var spec : specs) {
+            var cc = new CodegenConfigurator()
+                    .setInputSpec(spec.openApiSpec())
+                    .setLang(CrdtoolsCodegen.class.getCanonicalName())
+                    .setOutputDir(tmpOutputDir.toAbsolutePath().toString())
+                    .setModelPackage("com.gs.crdtools")
+                    // CodegenConfigurator modifies its Map arguments, so we need to wrap it in something mutable
+                    .setAdditionalProperties(
+                            HashMap.of(
+                                    "java8", (Object) true,
+                                    "hideGenerationTimestamp", true,
+                                    "notNullJacksonAnnotation", true
+                            ).toJavaMap()
+                    )
+                    .setTypeMappings(
+                            HashMap.of(
+                                    V1ObjectMeta.class.getSimpleName(),
+                                    V1ObjectMeta.class.getCanonicalName()
+                            ).toJavaMap()
+                    );
+
             new DefaultGenerator().opts(cc.toClientOptInput()).generate();
-            return readDir(tmpOutputDir, ".java");
-        } finally {
-            delete(tmpOutputDir);
+            result = result.merge(readDir(tmpOutputDir, ".java"));
         }
+
+        delete(tmpOutputDir);
```

Comment:
Don't remove the try {} finally, as it handles file deletion when there is an exception as well as when everything succeeds.